apps/web-giddh/src/app/shared/bank-integration/bank-link/bank-link.component.ts

```diff
@@ -0,0 +1,126 @@
+})
+
+export class BankLinkComponent implements OnInit {
+    /* This will hold local JSON data */
+    public localeData: any = {};
+    /* This will hold common JSON data */
+    public commonLocaleData: any = {};
+    /** True if api call in progress */
+    public isLoading: boolean = false;
+    /** List of connected bank accounts */
+    public connectedBankAccounts: any[] = [];
+    /** Hold selectedBank */
+    public selectedBanksList: string = '';
+    private destroyed$: ReplaySubject<boolean> = new ReplaySubject(1);
+    /** Hold options of dropdown  */
+    public bankLinks: IOption[] = []
+    /** Hold selected bank from dropdown/bankLinks */
+    public selectedBank: any;
+    /** Hold all connected bank */
+    public bankAccounts: any[] = [];
+    /** True if relogin required in any bank account */
+    public reLoginRequired: boolean = false;
+
+    constructor(
+        private settingsIntegrationService: SettingsIntegrationService,
+        private toasty: ToasterService,
+        public dialog: MatDialog,
+        public dialogRef: MatDialogRef<BankLinkComponent>,
+        @Inject(MAT_DIALOG_DATA) public inputData
+    ) { }
+    /**
+     * Initializes the component
+     *
+     * @memberof BankLinkComponent
+     */
+    public ngOnInit(): void {
+        if (!this.inputData.bankList) {
+            this.getAllBankAccounts();
+        } else {
+            this.setTransformBankListData(this.inputData.bankList);
+        }
+    }
+    /**
+     * This will use for select bank account
+     * 
+     * @param {*} event
+     * @memberof BankLinkComponent
+     */
+    public selectedOption(event: any): void {
+        if (event?.additional) {
+            this.selectedBank = event.additional
+        }
+    }
+    /**
+     * This will link all the connected bank accounts
+     * 
+     * @memberof BankLinkComponent
+     */
+    public linkBankAccount(): void {
+        let request = { bankAccountUniqueName: this.selectedBank?.bankResource?.uniqueName };
+        let accountForm = {
+            accountNumber: this.selectedBank?.bankResource?.accountNumber,
+            accountUniqueName: this.inputData?.accountUniqueName,
+            paymentAlerts: []
+        };
+        this.settingsIntegrationService.updateAccount(accountForm, request)
+            .pipe(takeUntil(this.destroyed$))
+            .subscribe({
+                next: (response) => {
+                    if (response?.status === "success") {
+                        if (response?.body?.message) {
+                            this.toasty.clearAllToaster();
+                            this.toasty.showSnackBar('success', response?.body?.message);
+                            this.dialogRef.close();
+                        }
+                    }
+                },
+                error: (error) => {
+                    this.toasty.clearAllToaster();
+                    this.toasty.showSnackBar('error', error?.message || 'Something went wrong');
```

Comment:
this.commonLocaleData?.app_validations?.something_went_wrong

Reply:
done